includes/collection/class-outbox.php

```diff
@@ -296,19 +289,44 @@ public static function maybe_get_activity( $outbox_item ) {
 		return self::get_activity( $outbox_item );
 	}
 
+	/**
+	 * Get the object ID of an activity.
+	 *
+	 * @param Activity $activity The activity object.
+	 * @return string The object ID.
+	 */
+	private static function get_object_id( $activity ) {
+		// Most common.
+		if ( is_object( $activity->get_object() ) ) {
+			return $activity->get_object()->get_id();
+		}
+
+		// Rare.
+		if ( is_string( $activity->get_object() ) ) {
+			return $activity->get_object();
+		}
+
+		// Exceptional.
+		return $activity->get_actor() ?? $activity->get_id();
+	}
+
 	/**
 	 * Get the title of an activity recursively.
 	 *
 	 * @param \Activitypub\Activity\Base_Object $activity_object The activity object.
 	 * @return string The title.
 	 */
-	private static function recursively_get_title( $activity_object ) {
+	private static function get_object_title( $activity_object ) {
 		if ( is_string( $activity_object ) ) {
 			$post_id = url_to_postid( $activity_object );
 
 			return $post_id ? get_the_title( $post_id ) : '';
 		}
 
+		if ( null === $activity_object ) {
```

Comment:
I would make this check more general and use it to bail early:

```PHP
if ( ! $activity_object ) { 
	return '';
}
```

Reply:
Fixed in [794e231](https://github.com/Automattic/wordpress-activitypub/pull/1474/commits/794e2316b52a17896a09102dcfe3e013702869e2)